onnxruntime/lora/adapter_format/adapter_schema.fbs

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace onnxruntime.adapters;
+
+// Tensor
+enum TensorDataType : int32 {
```

Comment:
You can avoid redefining this by using the existing definition inside ort.fbs: https://github.com/microsoft/onnxruntime/blob/d4692835bf7affb51afa83b47159f43e7526ac9d/onnxruntime/core/flatbuffers/schema/ort.fbs#L46

Example of how to reuse the existing fbs: https://github.com/microsoft/onnxruntime/blob/d4692835bf7affb51afa83b47159f43e7526ac9d/onnxruntime/core/flatbuffers/schema/ort_training_checkpoint.fbs#L4

Reply:
The adapters format is designed to be standardized and possibly moved to ONNX.
As such it must not be a blackbox for end users but must be publicly defined.
It cannot depend on ORT private format which as subject to change release to release. Said that, the only thing which potentially can be shared in type enums.
It also does not share the namespace which was decided upon by PMs.
For the above reasons it does not and cannot share any utility code, which I checked, amounts only to a few lines of code (Save/Load string) everything else is different.